modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1537,7 +1540,7 @@ static CompletableFuture<List<BinaryRow>> collectMultiRowsResponsesWithRestoreOr
                     }
 
                     // Use Arrays#asList to avoid copying the array.
-                    return Arrays.asList(result);
+                    return skipNull ? Arrays.stream(result).filter(r -> r != null).collect(Collectors.toList()) : Arrays.asList(result);
```

Comment:
Please don't use the stream.

Reply:
Let's leave it as is. We will be able to chage it if it would light up in performance benchmarks.

Reply:
But why wait for benchmarks?

Reply:
Because any change should be motivated.

Reply:
This piece of code is a candidate for the hot path, we need to fix it now.

Reply:
If we had a benchmarc that proves your proposal.
Currently, this code looks right to me.

Reply:
I would rather skip given extra null filtering here and add null checks in a few lines above where we add resultRow to a batch. Follow up trimming using `Arrays.copyOf()` is also required if skipNull and original size != i.

Reply:
It is not possible to skip null somewhere in the method because the result array is formed by the order of rows in the replication request. Result array has the same order as it is in the request, but for some operations (insertAll, deleteAll, e.t.c.), we should provide only rows that were not applied to replications. Hence, we are skipping the appropriate rows in the array in the correct order.

Reply:
Do you mean that 
```
                        for (BinaryRow resultRow : batchResult) {
                            if (skipNull) {
                                if (resultRow == null) {
                                    // Do nothing
                                } else {
                                    result[rowBatch.getOriginalRowIndex(i++)] = resultRow;
                                }
                            } else {
                                result[rowBatch.getOriginalRowIndex(i++)] = resultRow;    
                            }
                        }
```
won't work?

Reply:
Got it. Not possible. Please ignore.